lib/r10k/source/exec.rb

```diff
@@ -0,0 +1,53 @@
+require 'r10k/util/subprocess'
+require 'json'
+require 'yaml'
+
+class R10K::Source::Exec < R10K::Source::Hash
+  R10K::Source.register(:exec, self)
+
+  def initialize(name, basedir, options = {})
+    unless @command = options[:command]
+      raise ConfigError, _('Environment source %{name} missing required parameter: command') % {name: name}
+    end
+
+    # We haven't set the environments option yet. We will do that by
+    # overloading the #environments method.
+    super(name, basedir, options)
+  end
+
+  def environments
+    if @environments.nil?
+      set_environments_hash(run_environments_command)
+      super
+    end
+    @environments
+  end
+
+  def run_environments_command
```

Comment:
Non-blocker: this method should be private or protected.

Reply:
Made method private.